x-pack/plugins/infra/public/hooks/use_kibana.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { CoreStart } from '../../../../../src/core/public';
+import {
+  createKibanaReactContext,
+  KibanaReactContextValue,
+  useKibana,
+} from '../../../../../src/plugins/kibana_react/public';
+import { InfraClientStartDeps } from '../types';
+
+export type PluginKibanaContextValue = CoreStart & InfraClientStartDeps;
+
+export const createKibanaContextForPlugin = (core: CoreStart, pluginsStart: InfraClientStartDeps) =>
```

Comment:
:information_source: This in combination with the `useKibanaContextForPlugin()` hook below aims at providing a typed Kibana context value as recommended by the `kibana_react` plugin.